### PR TITLE
add clear user param to disconnect

### PR DESCRIPTION
**Pull request by imtoori**
If true dispose the client state and instantiate a new one

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -793,12 +793,20 @@ class Client {
   }
 
   /// Closes the websocket connection and resets the client
```
Comment (Nash0x7E2) [inline lib/src/client.dart:795]: I think we should update the documentation for this method as well.

File: lib/src/client.dart [hunk 2]
```diff
@@ -793,12 +793,20 @@ class Client {
...
+  Future<void> disconnect({
+    bool flushOfflineStorage = false,
+    bool clearUser = false,
+  }) async {
     logger.info('Disconnecting');
 
     await _offlineStorage?.disconnect(flush: flushOfflineStorage);
     _offlineStorage = null;
 
+    if (clearUser == true) {
```
Comment (Nash0x7E2) [inline lib/src/client.dart:805]: What about a log here? Might be useful for debugging purposes. At the very least, we can log the options the function was called with.

**Review by Nash0x7E2 — CHANGES_REQUESTED** [review]
Left some comments @imtoori. What do you think?

**Comment by vishalnarkhede** [comment]
@mahboubii we should do the same later for js client

```js
client.disconnect({ clearUser: false }):
client.connect()
```

is much better than

```js
client.wsConnection.disconnect() 
client.setupConnection()

// (this is what we recommend to customers right now)
```

for disconnecting and reconnecting experience, which is mostly used for foregrounding/background the application.

What do you think?

**Comment by mahboubii** [comment]
> @mahboubii we should do the same later for js client
> 
> ```js
> client.disconnect({ clearUser: false }):
> client.connect()
> ```
> 
> is much better than
> 
> ```js
> client.wsConnection.disconnect() 
> client.setupConnection()
> 
> // (this is what we recommend to customers right now)
> ```
> 
> for disconnecting and reconnecting experience, which is mostly used for foregrounding/background the application.
> 
> What do you think?

yea sounds good to me, we need to do some work around all the WS stuff

**Review by Nash0x7E2 — APPROVED** [review]
LGTM

**Pull request merged**